### ISSUE TITLE
Optimization problem for DPUCADF8H (Vitis AI 2.0)

### ISSUE THREAD

**Issue by cl600class** [Xilinx/Vitis-AI#745]
I'm working on a [QuartzNet](https://arxiv.org/abs/1910.10261)-based model (includes depthwise convolutions) with Pytorch, and it is successfully deployed on `ZCU102` by modifying the [tutorial](https://github.com/Xilinx/Vitis-AI-Tutorials/tree/master/Design_Tutorials/09-mnist_pyt).

To deploy the model using DPUCADF8H DPUs, I replaced the depthwise & pointwise convolutions with the normal ones.
For example: (the input was resized to (N, C, H, W) = (batchsize, 1, chunk_length, 1))
> \# nn.Conv2d(n_channel, n_channel, kernel_size=(n_kernel,1), stride=(1,1), padding=(n_padding,0), groups=n_channel, bias=False),
\# nn.Conv2d(n_channel, n_channel, kernel_size=(1,1), stride=(1,1), bias=False),
nn.Conv2d(n_channel, n_channel, kernel_size=(n_kernel,1), stride=(1,1), padding=(n_padding,0), bias=False),

However, there is an error during the compilation stage `Code Generation at Node Level and then Recursively` with Vitis AI 2.0.

![image](https://user-images.githubusercontent.com/31886974/161811866-da941602-cffa-4bb3-831d-13363a623598.png)

According to the error message, I found there are some optimization options in the function `compile_sc_dpuv3int8_default` of `/opt/vitis_ai/conda/envs/vitis-ai-pytorch/lib/python3.6/site-packages/SC/compiler_func.py`

![image](https://user-images.githubusercontent.com/31886974/161813514-c5ac7887-987f-423d-a301-160e3972b616.png)

If I assign `compiler_args.operation_fusion_elt` to `False` in line 74, the model can be compiled without any error.

The generated graph can be viewed [here](https://i.imgur.com/q1qJUTf.png). There is no depthwise convolution.

I have deployed it to both Azure and AWS F1 instances based on the same tutorial with version 1.4, but both produced a segmentation fault on step `dpu.execute_async()`.

Since using different versions of Vitis AI for inferencing & compilation is feasible according to #736, the problem may be attributed to the optimization options mentioned above (especially eltwise layers).

I would like to know how they affect the generated xmodel and if there is any suggestion about the issue. Thanks.

**Comment by cl600class** [comment]
Update:

1. The residual blocks take the wrong inputs which makes `compiler_args.operation_fusion_elt` unable to handle this condition. After modifying the model, the compilation is successful with default optimization configurations. 
```
def forward(self, x):
    o1 = self.conv(x)
    o2 = self.residual(x)    # The input was falsely set to 'o1'
    return self.activation(o1 + o2)
```
2. The segmentation fault problem on step `dpu.execute_async()` is due to the I/O buffer type. In version 1.4, the type should be a list containing a numpy array.
